system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -3908,6 +3908,11 @@ class CLdapClient : public CInterface, implements ILdapClient
         StringBuffer filter;
         filter.append("distinguishedName=").append(dn);
 
+        filter.replaceString("\\", "\\5c");//Replace special characters with valid UTF-8 string (see valueencoding rule in RFC 4515)
+        filter.replaceString("*", "\\2a");
+        filter.replaceString("(", "\\28");
+        filter.replaceString(")", "\\29");
```

Comment:
encodings.h is not part of the platform. However, this is an LDAP specific this so this is probably the best place for it